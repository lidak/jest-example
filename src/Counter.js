import React, { Component } from 'react';

export default class Counter extends Component {
  constructor({number}) {
    super();
    this.state = {
      number
    };
  }

  increaseNumber() {
    this.setState({
      number: ++this.state.number
    })
  }

  getMessage(number) {
    if(number === 0) {
      return 'zero';
    } else if(number === 1) {
      return `Button clicked ${number} time.`
    } else {
      return `Button was clicked ${number} times!!!`;
    }
  }

  render() {
    const {
      number
    } = this.state;
    const message = this.getMessage(number);

    return (
      <div className="counter">
        <div className="number">{message}</div>
        <button onClick={this.increaseNumber.bind(this)}>+</button>
      </div>
    )
  }
}