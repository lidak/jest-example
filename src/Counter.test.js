import React from 'react';
import Counter from './Counter';
import {mount, shallow} from 'enzyme';
import renderer from 'react-test-renderer';

describe('Counter component', () => {
  const tree = renderer
    .create(<Counter number={2} />)
    .toJSON();

  test('should be rendered correctly', () => {
    expect(tree).toMatchSnapshot();
  });

  test('number should be increased in DOM after + is clicked', () => {
    const counterComponent = mount(<Counter number={2} />);
    const button = counterComponent.find('button');
    const numberContainer = counterComponent.find('.number');

    expect(numberContainer.text()).toEqual('Button was clicked 2 times!!!');

    button.simulate('click');

    expect(numberContainer.text()).toEqual('Button was clicked 3 times!!!');
  });

  test('should return appropriate messages for different numbers', () => {
    const counter = shallow(<Counter number={0}/>).instance();

    expect(counter.getMessage(0)).toEqual('zero');
    expect(counter.getMessage(1)).toEqual('Button clicked 1 time.');
    expect(counter.getMessage(999)).toEqual('Button was clicked 999 times!!!');
  });
});

