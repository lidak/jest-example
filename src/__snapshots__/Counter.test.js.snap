// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter component should be rendered correctly 1`] = `
<div
  className="counter"
>
  <div
    className="number"
  >
    Button was clicked 2 times!!!
  </div>
  <button
    onClick={[Function]}
  >
    +
  </button>
</div>
`;
